File: src/main/java/org/wcardinal/controller/data/internal/SContainers.java
/*
 * Copyright (C) 2019 Toshiba Corporation
 * SPDX-License-Identifier: Apache-2.0
 */

package org.wcardinal.controller.data.internal;

public class SContainers {
	private final static String ID_PREFIX = "$$";

	public static String toId( final String name, final String type ) {
		return ID_PREFIX+name+"@"+type;
	}

	public static boolean isId( final String name ) {
		return name.startsWith( ID_PREFIX );
	}

	public static boolean isNotId( final String name ) {
		return ! isId( name );
	}
}

File: src/main/typescript/wcardinal/controller/data/internal/s-containers.ts
/*
 * Copyright (C) 2019 Toshiba Corporation
 * SPDX-License-Identifier: Apache-2.0
 */

const SCONTAINER_ID_PREFIX = "$$";
const SCONTAINER_ID_CHAR = "$";

export const toSContainerId = ( name: string, type: string ): string => {
	return `${SCONTAINER_ID_PREFIX}${name}@${type}`;
};

export const isSContainerId = ( name: string ): boolean => {
	return name[ 0 ] === SCONTAINER_ID_CHAR && name[ 1 ] === SCONTAINER_ID_CHAR;
};

export const isNotSContainerId = ( name: string ): boolean => {
	return ! isSContainerId( name );
};
